.golangci.yml

```diff
@@ -73,6 +73,60 @@ linters-settings:
           - pkg: "crypto/md5"
           - pkg: "crypto/sha1"
           - pkg: "crypto/**/pkix"
+      otlp-internal:
+        files:
+          # Do not allow even the 
+          - "**/otel/exporters/otlp/**/*.go"
+          # TODO: remove the following when otlpmetric/internal is removed.
+          - "!**/otel/exporters/otlp/otlpmetric/internal/oconf/envconfig.go"
+          - "!**/otel/exporters/otlp/otlpmetric/internal/oconf/options.go"
+          - "!**/otel/exporters/otlp/otlpmetric/internal/oconf/options_test.go"
+          - "!**/otel/exporters/otlp/otlpmetric/internal/otest/client_test.go"
+        deny:
+          - pkg: "go.opentelemetry.io/otel/exporters/otlp/internal"
+            desc: Do not use cross-module internal packages.
+      otlptrace-internal:
+        files:
+          - "**/otel/exporters/otlp/otlptrace/**/*.go"
+          - "!**/otel/exporters/otlp/otlptrace/internal/*.go"
+          - "!**/otel/exporters/otlp/otlptrace/internal/**/*.go"
```

Comment:
I think that the depguard rules should look more or les like

```suggestion
      otlptrace-internal:
        files:
          - "!**/exporters/otlp/otlptrace/*.go"
          - "!**/exporters/otlp/otlptrace/internal/**.go"
```

1. `- "**/otel/exporters/otlp/otlptrace/**/*.go"` becasue of this we can miss something
2. I think that `otel` should not be in part of "file path"
3. Additional fine-grained exceptions can be added like this: `import "go.opentelemetry.io/otel/exporters/otlp/otlpmetric/internal/transform" //nolint:staticcheck,depguard // Atomic deprecation.`


The same comment applies to all rules (with exception to `otel-internal:`)

Here is what I created

```yaml
      otlp-internal:
        files:
          - "!**/exporters/otlp/internal/**/*.go"
          # TODO: remove the following when otlpmetric/internal is removed.
          - "!**/exporters/otlp/otlpmetric/internal/oconf/envconfig.go"
          - "!**/exporters/otlp/otlpmetric/internal/oconf/options.go"
          - "!**/exporters/otlp/otlpmetric/internal/oconf/options_test.go"
          - "!**/exporters/otlp/otlpmetric/internal/otest/client_test.go"
        deny:
          - pkg: "go.opentelemetry.io/otel/exporters/otlp/internal"
            desc: Do not use cross-module internal packages.
      otlptrace-internal:
        files:
          - "!**/exporters/otlp/otlptrace/*.go"
          - "!**/exporters/otlp/otlptrace/internal/**.go"
        deny:
          - pkg: "go.opentelemetry.io/otel/exporters/otlp/otlptrace/internal"
            desc: Do not use cross-module internal packages.
      otlpmetric-internal:
        files:
          - "!**/exporters/otlp/otlpmetric/internal/*.go"
          - "!**/exporters/otlp/otlpmetric/internal/**/*.go"
        deny:
          - pkg: "go.opentelemetry.io/otel/exporters/otlp/otlpmetric/internal"
            desc: Do not use cross-module internal packages.
      otel-internal:
        files:
          - "**/sdk/*.go"
          - "**/sdk/**/*.go"
          - "**/exporters/*.go"
          - "**/exporters/**/*.go"
          - "**/schema/*.go"
          - "**/schema/**/*.go"
          - "**/metric/*.go"
          - "**/metric/**/*.go"
          - "**/bridge/*.go"
          - "**/bridge/**/*.go"
          - "**/example/*.go"
          - "**/example/**/*.go"
          - "**/trace/*.go"
          - "**/trace/**/*.go"
        deny:
          - pkg: "go.opentelemetry.io/otel/internal$"
            desc: Do not use cross-module internal packages.
          - pkg: "go.opentelemetry.io/otel/internal/attribute"
            desc: Do not use cross-module internal packages.
          # TODO: Blocked on #4424
          #- pkg: "go.opentelemetry.io/otel/internal/internaltest"
          #  desc: Do not use cross-module internal packages.
          - pkg: "go.opentelemetry.io/otel/internal/matchers"
            desc: Do not use cross-module internal packages.
```